### PR TITLE
Elg 266

### PR DESCRIPTION
**Pull request by gcleary**
Renamed beacon_in_cache to beacon_in_cacheline, and implemented beacon_in_cache which does search all cachelines. This code should be no more expensive for the CACHE_SIZE 1 case, but should correctly pick up Used AP info from any cacheline making cache match more efficient when CACHE_SIZE is greater than 1.

I will create specific unit_tests for this case along with a comprehensive set of unit_tests.

### CODE REVIEW

File: libel/protocol/Makefile [hunk 1]
```diff
@@ -3,6 +3,8 @@ include ../../.submodules/nanopb/extra/nanopb.mk
 
 TARGETS = el.pb.c el.pb.h
 
+PROTOC_OPTS = -I.
+
```
Comment (tboinske-skyhook) [inline libel/protocol/Makefile:7]: This line causes the build to break for me:
```
$ make clean;make
make -C libel/protocol clean
rm -f el.pb.c el.pb.h
rm -rf bin build
mkdir -p bin
mkdir -p build
make -C libel/protocol
protoc -I. --nanopb_out=. el.proto
protoc-gen-nanopb: program not found or is not executable
Please specify a program using absolute path or make sure the program is available in your PATH system variable
--nanopb_out: protoc-gen-nanopb: Plugin failed with status code 1.
make[1]: *** [el.pb.c] Error 1
make: *** [libel/protocol/el.pb.h] Error 2
```
If I remove it, the build works fine. What motivated you to add it?

File: libel/beacons.c [hunk 2]
```diff
@@ -442,121 +571,235 @@ static bool beacon_in_cache(Sky_ctx_t *ctx, Beacon_t *b, Sky_cacheline_t *cl)
...
+            break;
+    }
+    if (index)
+        *index = (ret == true) ? j : -1;
     return ret;
 }
 
-/*! \brief find cache entry with best match
- *  if beacon is AP, filter
+/*! \brief count number of cached APs in workspace relative to a cacheline
```
[the remark this answers is not in the document]
Comment (gcleary) [reply libel/beacons.c]: Yes, Done. Changed count_used_aps to be consistent

File: libel/beacons.c [hunk 3]
```diff
@@ -371,30 +455,75 @@ Sky_status_t add_beacon(Sky_ctx_t *ctx, Sky_errno_t *sky_errno, Beacon_t *b, boo
     return sky_return(sky_errno, SKY_ERROR_NONE);
 }
 
+/*! \brief check if a beacon is in a cache
+ *
+ *   Scan all beacons in the cache.
+ *   the appropriate attributes. If the given beacon is found in the cacheline,
```
[the remark this answers is not in the document]
Comment (gcleary) [reply libel/beacons.c]: Yes. Done.

File: libel/protocol/Makefile (hunk 1 above)
Comment (gcleary) [reply libel/protocol/Makefile]: This was mistakenly merged from my ELG-263 work. I was temporarily overriding some files there and never should have committed that.

**Comment by gcleary** [comment]
The last commit to beacons.c fixes a problem in my implementation of beacon_in_cache. It was aborting the search if an empty cacheline was tested. Also was not consistently returning -1 in the indexes if no match was found.

**Comment by gcleary** [comment]
This branch has been migrated [here](https://github.com/SkyhookWireless/embedded-client-private/tree/EC-66)

**Pull request closed without merging**